daemon/cluster/executor/container/container.go

```diff
@@ -598,16 +598,19 @@ func (c *containerConfig) networkCreateRequest(name string) (clustertypes.Networ
 		}
 	}
 
-	if na.Network.DriverState != nil {
-		options.Driver = na.Network.DriverState.Name
-		options.Options = na.Network.DriverState.Options
+	driverConfig := na.Network.Spec.DriverConfig
```

Comment:
Not sure if I am missing something. But spec will only have it if the network was created with the options. If subnet is not specified during network create, then swam will allocate a subnet in which case the spec will not have it.

Reply:
I think the right fix would be check on how the network attachment is created for bridge network. FWIK this change will break network creation for overlay networks atleast.

Reply:
Ah thanks @abhi for your explanations. That makes sense, I will take a look into swarmkit tomorrow and try to investigate the problem there. May I ping you over there if I'm able to open a PR?

I'm going to close this one here.

Reply:
@stffabi absolutely.